src/NHibernate/Impl/MultiQueryImpl.cs

```diff
@@ -458,30 +458,37 @@ protected void After()
 
 		protected virtual IList GetResultList(IList results)
 		{
-			var resultCollections = new List<object>(resultCollectionGenericType.Count);
-			for (int i = 0; i < queries.Count; i++)
+			var rawResultCollections = new List<IList>(resultCollectionGenericType.Count);
+			for (var i = 0; i < queries.Count; i++)
 			{
-				if (resultCollectionGenericType[i] == typeof(object))
-				{
-					resultCollections.Add(new List<object>());
-				}
-				else
-				{
-					resultCollections.Add(Activator.CreateInstance(typeof(List<>).MakeGenericType(resultCollectionGenericType[i])));
-				}
+				var query = queries[i] as ExpressionQueryImpl;
+				// Linq queries may override the query type, finishing the work with a post execute transformer,
+				// which with multi queries are executed through the multi-query result trasformer.
+				var rawElementType = query?.QueryExpression?.Type ?? resultCollectionGenericType[i];
```

Comment:
Grabbing here the Linq result type override if any, for typing the list before transformation with nullable if required.